core/src/main/java/org/apache/carbondata/core/carbon/datastore/block/TableBlockInfo.java

```diff
@@ -81,13 +84,14 @@ public TableBlockInfo(String filePath, long blockOffset, String segmentId, Strin
    * @param blockletInfos
    */
   public TableBlockInfo(String filePath, long blockOffset, String segmentId, String[] locations,
-      long blockLength, BlockletInfos blockletInfos) {
+      long blockLength, BlockletInfos blockletInfos, short version) {
     this.filePath = FileFactory.getUpdatedFilePath(filePath);
     this.blockOffset = blockOffset;
     this.segmentId = segmentId;
     this.locations = locations;
     this.blockLength = blockLength;
     this.blockletInfos = blockletInfos;
+    this.version = version;
```

Comment:
validateCarbonDataFileVersion method is CrabonProperties I am validating the version